plugin-vidarr/src/main/java/ca/on/oicr/gsi/shesmu/vidarr/VidarrPlugin.java

```diff
@@ -50,6 +54,99 @@
 
 public class VidarrPlugin extends JsonPluginFile<Configuration> {
 
+  private class ProvenanceCache extends ValueCache<Stream<VidarrAnalysisValue>> {
+    public ProvenanceCache(Path fileName) {
+      super("vidarr-provenance " + fileName.toString(), 30, ReplacingRecord::new);
+    }
+
+    private String formatListAsString(List<String> list) {
+      return list.stream().map(name -> ("\"" + name + "\"")).collect(Collectors.joining(","));
+    }
+
+    private String createVidarrProvenanceRequestBody(List<String> versionTypes) {
+      return "{"
+          + "\"analysisTypes\": [\"FILE\"],"
+          + "\"epoch\": 0,"
+          + "\"includeParameters\": false,"
+          + "\"timestamp\": 0,"
+          + "\"versionPolicy\": \"LATEST\","
+          + "\"versionTypes\": ["
+          + versionTypes.stream().map(name -> ("\"" + name + "\"")).collect(Collectors.joining(","))
+          + "  ]}";
+    }
+
+    protected Stream<VidarrAnalysisValue> provenanceArchive(String baseUrl) throws Exception {
+
+      if (configuration.isEmpty()) {
```

Comment:
the only place provenanceArchive() is called is in fetch(), which also tests for empty configuration. Don't need to test twice